figures/tasks.py

```diff
@@ -258,6 +263,64 @@ def populate_daily_metrics(site_id=None, date_for=None, force_update=False):
                            site_count=sites_count))
 
 
+@shared_task
+def populate_daily_metrics_next(site_id=None, force_update=False):
+    """Next iteration to collect daily metrics for all sites in a deployment
+
+    This is a top level Celery task run every 24 hours to update Figures data.
+
+    * It updates Figures per-enrollment data and collect daily aggregate metrics
+    * It's purpose is to collect new metrics on an ongoing basis and not serve
+      dual purpose of collecting ongoing data AND backfilling data.
+    * The driver for this change is to improve performance of the daily Celery jobs
+
+    What's different?
+
+    * Figures collects the enrollment data first, then aggregates daily data.
+
+    TODO: Draft up public architecture docs and reference them here
+    """
+    if waffle.switch_is_active(WAFFLE_DISABLE_PIPELINE):
+        logger.warning('Figures pipeline is disabled due to %s being active.',
+                       WAFFLE_DISABLE_PIPELINE)
+        return
+
+    date_for = datetime.datetime.utcnow().date()
+    if site_id is not None:
+        sites = get_sites_by_id((site_id, ))
+    else:
+        sites = get_sites()
+    sites_count = sites.count()
+    # This is our task entry log message
+    msg = '{prefix}:START:date_for={date_for}, site_count={site_count}'
+    logger.info(msg.format(prefix=FPD_LOG_PREFIX,
+                           date_for=date_for,
+                           site_count=sites_count))
+    for i, site in enumerate(sites):
+        msg = '{prefix}:SITE:START:{id}:{domain} - Site {i:04d} of {n:04d}'
+        logger.info(msg.format(prefix=FPD_LOG_PREFIX,
+                               id=site.id,
+                               domain=site.domain,
+                               i=i,
+                               n=sites_count))
+        try:
+            populate_daily_metrics_for_site(site_id=site.id,
+                                            date_for=date_for,
+                                            ed_next=True,
+                                            force_update=force_update)
```

Comment:
Celery queue question: This would run `populate_daily_metrics_for_site` synchronously. Do we want this behavior?

`populate_daily_metrics_for_site.delay()` should be easier to work with in terms of deployments and queue impact.

We have the `group()` and `delay()` pattern used in figures:

https://github.com/appsembler/figures/blob/2f087967d5cb5d2246a8bcd6637a1614d8e1aebf/figures/tasks.py#L456-L468

Reply:
@OmarIthawi I _would_ prefer to run the tasks synchronously. We have not historically because when I originally implemented the celery tasks, I tried it synchronously with very simple vanilla tasks on standalone edx-platform and they would just "poof" disappear. No logs, no exceptions, nothing. Just drop out of existence. So we carried them on as serial executions since.

Soon in my queue is to drastically improve Figures backfill / data repair toolkit and health checking, and improve the Figures Celery docker devsite environment.

Once we have those plus documentation and training the platform team on how to use it. THEN we can optimize Figures tasks to run in parallel as you mention